backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h

```diff
@@ -1929,39 +1923,57 @@ template <typename Torus> struct int_sc_prop_memory {
           return does_overflow_if_carry_is_0;
         }
       };
-      auto overflow_flag_last = lut_overflow_flag_last->get_lut(0, 0);
+      auto overflow_flag_last = lut_message_extract->get_lut(0, 1);
 
       generate_device_accumulator<Torus>(
           streams[0], gpu_indexes[0], overflow_flag_last, glwe_dimension,
           polynomial_size, message_modulus, carry_modulus, f_overflow_last);
 
-      lut_overflow_flag_last->broadcast_lut(streams, gpu_indexes, 0);
+      Torus *h_lut_indexes =
+          (Torus *)malloc((num_radix_blocks + 1) * sizeof(Torus));
+      for (int index = 0; index < num_radix_blocks + 1; index++) {
+        if (index < num_radix_blocks) {
+          h_lut_indexes[index] = 0;
+        } else {
+          h_lut_indexes[index] = 1;
+        }
+      }
+      cuda_memcpy_async_to_gpu(
+          lut_message_extract->get_lut_indexes(0, 0), h_lut_indexes,
+          (num_radix_blocks + 1) * sizeof(Torus), streams[0], gpu_indexes[0]);
+
+      lut_message_extract->broadcast_lut(streams, gpu_indexes, 0);
+      free(h_lut_indexes);
     }
     if (requested_flag == outputFlag::FLAG_CARRY) { // Carry case
-      lut_carry_flag_last = new int_radix_lut<Torus>(
-          streams, gpu_indexes, gpu_count, params, 1, 1, allocate_gpu_memory);
 
       auto f_carry_last = [](Torus block) -> Torus {
         return ((block >> 2) & 1);
       };
-      auto carry_flag_last = lut_carry_flag_last->get_lut(0, 0);
+      auto carry_flag_last = lut_message_extract->get_lut(0, 1);
 
       generate_device_accumulator<Torus>(
           streams[0], gpu_indexes[0], carry_flag_last, glwe_dimension,
           polynomial_size, message_modulus, carry_modulus, f_carry_last);
 
-      lut_carry_flag_last->broadcast_lut(streams, gpu_indexes, 0);
+      Torus *h_lut_indexes =
+          (Torus *)malloc((num_radix_blocks + 1) * sizeof(Torus));
+      for (int index = 0; index < num_radix_blocks + 1; index++) {
+        if (index < num_radix_blocks) {
+          h_lut_indexes[index] = 0;
+        } else {
+          h_lut_indexes[index] = 1;
+        }
+      }
+      cuda_memcpy_async_to_gpu(
+          lut_message_extract->get_lut_indexes(0, 0), h_lut_indexes,
```

Comment:
should this be ->get_lut_indexes(0,1) ?

Reply:
No here I copy the full array so it's expected I get the lut_indexes of lut 0